src/mantle/framework/database/model/relations/class-belongs-to.php

```diff
@@ -52,17 +57,29 @@ public function __construct( Builder $query, Model $parent, string $foreign_key,
 	 * Add constraints to the query.
 	 */
 	public function add_constraints() {
-		$meta_value = $this->parent->get_meta( $this->local_key );
-
-		if ( empty( $meta_value ) ) {
-			/**
-			 * Prevent the query from going through.
-			 *
-			 * @todo Handle missing meta value better.
-			 */
-			$this->query->where( 'id', PHP_INT_MAX );
+		if ( $this->uses_terms ) {
+			$object_ids = $this->get_term_ids_for_relationship();
+
+			if ( empty( $object_ids ) ) {
+				// Prevent the query from going through.
+				// @todo Handle this better.
+				return $this->query->where( 'id', PHP_INT_MAX );
```

Comment:
(just curious, for my own info) is what makes it hard to handle the query like this that this function needs to return a query?

Reply:
In this case, we need to essentially kill the query. I would have thrown an exception, but I haven't set that up yet so this is the "easiest" way to kill a query.